### DISCUSSION TITLE
connect() with WalletConnectConnector giving Error: Chains must not be empty

### DISCUSSION THREAD

**Discussion by Mockafjongen** [wevm/wagmi#2398] [Q&A]
This is my code
`
import { arbitrum } from '@wagmi/core/chains';
import { publicProvider } from '@wagmi/core/providers/public'
import { infuraProvider } from '@wagmi/core/providers/infura'
import { MetaMaskConnector } from '@wagmi/core/connectors/metaMask';
import { WalletConnectConnector } from '@wagmi/core/connectors/walletConnect'
import { configureChains, connect, createConfig, fetchBalance } from '@wagmi/core';

const projectId = '...';
const chains = [arbitrum];
const { publicClient, webSocketPublicClient } = configureChains(chains, [
	infuraProvider({apiKey: "..."}),
	publicProvider()
]);
const walletConnectConnector = new WalletConnectConnector({
	options: {
		projectId,
	},
	chains
});
const wagmiConfig = createConfig({
	autoConnect: true,
	connectors: walletConnectConnector,
	publicClient,
	webSocketPublicClient
});

async function connectWalletConnect() {
	const result = await connect({
		connector: walletConnectConnector
	});
}
`

but when i execute connectWalletConnect() i get Error: Chains must not be empty from walletConnect.js:90
how can i fix this?

I'm using SvelteKit

**Comment by SaadNiaziDev** [comment]
this error occurs when we use any chain (only one) other that mainnet. if you include mainnet in chains array, this error will go away 
``
const chains = [mainnet,arbitrum];
``

don't know why this happens 😄

**Reply by agentx3** [reply]
Wow thanks so much. In my case, this happened when I was using only 
`chains = [mainnet]`. So I had to add another chain (that I don't even use) 
`chains = [mainnet, optimism]` to resolve the issue. So weird.
